tests/static/js/formatters-internal/image.js

```diff
@@ -1,106 +1,197 @@
+      photoImgUrl = Formatters.image(photoImg, 'x500', true).url;
+      expect(photoImgUrl).toEqual('https://dyn.mktgcdn.com/p/FOO/fit=contain,height=500');
+      photoWithPaddingImgUrl = Formatters.image(photoWithPaddingImg, 'x500', true).url;
+      expect(photoWithPaddingImgUrl).toEqual('https://dyn.mktgcdn.com/p/FOO/1.0000/fit=contain,height=500');
+      oldFileImgUrl = Formatters.image(oldFileImg, 'x500', true).url;
+      expect(oldFileImgUrl).toEqual('https://dyn.mktgcdn.com/f/0/FOO.jpg/fit=contain,height=500');
+      newFileImgUrl = Formatters.image(newFileImg, 'x2000', true).url;
+      expect(newFileImgUrl).toEqual('https://dyn.mktgcdn.com/f/FOO.jpg/fit=contain,height=2000');
+      euFileImgUrl = Formatters.image(euFileImg, 'x2000', true).url;
+      expect(euFileImgUrl).toEqual('https://dyn.eu.mktgcdn.com/f/0/FOO.jpg/fit=contain,height=2000');
     });
   });
 
-  describe('when image is served from EU with no dimensions specified', () => {
-    it('when choosing the biggest image under threshold, use the width and height on the image object if exists', () => {
-      const euImageUrl = Formatters.image(
-        { url: euUrl,
-          width: 1024,
-          height: 768,
-          sourceUrl: 'https://a.mktgcdn.com/p/FOO/516x384.jpg',
-        }, 'x', false).url;
-      expect(euImageUrl).toEqual('https://dyn.eu.mktgcdn.com/f/0/FOO.jpg/width=1024,height=768,fit=contain');
+  describe('when image has no dimensions specified', () => {
+    it('by default chooses the smallest image with width >= 200', () => {
+      const photoImgUrl = Formatters.image(photoImg).url;
+      expect(photoImgUrl).toEqual('https://dyn.mktgcdn.com/p/FOO/fit=contain,width=200');
+      const photoWithPaddingImgUrl = Formatters.image(photoWithPaddingImg).url;
+      expect(photoWithPaddingImgUrl).toEqual('https://dyn.mktgcdn.com/p/FOO/1.0000/fit=contain,width=200');
+      const oldFileImgUrl = Formatters.image(oldFileImg).url;
+      expect(oldFileImgUrl).toEqual('https://dyn.mktgcdn.com/f/0/FOO.jpg/fit=contain,width=200');
+      const newFileImgUrl = Formatters.image(newFileImg).url;
+      expect(newFileImgUrl).toEqual('https://dyn.mktgcdn.com/f/FOO.jpg/fit=contain,width=200');
+      const euFileImgUrl = Formatters.image(euFileImg).url;
+      expect(euFileImgUrl).toEqual('https://dyn.eu.mktgcdn.com/f/0/FOO.jpg/fit=contain,width=200');
     });
 
-    it('when choosing the biggest image under threshold, use dimensions from the sourceUrl if width/height does not exist', () => {
-      const euImageUrl = Formatters.image(
-        { url: euUrl,
-          width: 1024,
-          sourceUrl: 'https://a.mktgcdn.com/p/FOO/516x384.jpg'
-        }, 'x', false).url;
-      expect(euImageUrl).toEqual('https://dyn.eu.mktgcdn.com/f/0/FOO.jpg/width=516,height=384,fit=contain');
-    });
+    it('do not transform image regardless of atLeastAsLarge\'s', () => {
```

Comment:
can you update this test name to indicate that non-parseable or NaN dimensions are passed in? and can you also add tests for `desiredSize=''` and if `0` is passed for one or both desired dims?

Reply:
Done. desiredSize='' is forbidden from the caller